File: swordOffer/dp/numWays.cc
#include <iostream>
#include <vector>

using namespace std;

class Solution {
 public:
  int numWays(int n ) {

    if (n < 2) {
      return 1;
    }

    int n1 = 1;
    int n2 = 1;
    int res = 0;

    for (int i = 2; i <= n; i++) {
      res = (n1 + n2) % 1000000007;
      n1 = n2;
      n2 = res;
    }

    return res;
  }
};

File: swordOffer/go/dp/numWays.go
package main

func numWays(n int) int {
	if n == 1 || n == 0 {
		return 1
	}

	dp := make([]int, n+1)
	dp[0] = 1
	dp[1] = 1
	for i := 2; i <= n; i++ {
		dp[i] = (dp[i-1] + dp[i-2]) % 1000000007
	}

	return dp[len(dp)-1]
}
